test/test_urdf.py

```diff
@@ -181,6 +181,77 @@ def test_link_multiple_collision(self):
 </robot>'''
         self.parse_and_compare(xml)
 
+    def test_version_attribute_not_enough_dots(self):
+        xml = '''<?xml version="1.0"?>
+<robot name="test" version="1">
+</robot>'''
+        self.assertRaises(Exception, self.parse, xml)
+
+    def test_version_attribute_too_many_dots(self):
+        xml = '''<?xml version="1.0"?>
+<robot name="test" version="1.0.0">
+</robot>'''
+        self.assertRaises(Exception, self.parse, xml)
+
+    def test_version_attribute_not_enough_numbers(self):
+        xml = '''<?xml version="1.0"?>
+<robot name="test" version="1.">
+</robot>'''
+        self.assertRaises(Exception, self.parse, xml)
+
+    def test_version_attribute_no_major_number(self):
+        xml = '''<?xml version="1.0"?>
+<robot name="test" version="1.">
```

Comment:
for `no_major_number`, should this be `version=".0"`?

Reply:
Yeah, definitely.  That was a mistake, and it actually showed a bug.  Will fix.